### PR TITLE
fix for 6340 bug

### PR DESCRIPTION
**Pull request by praneethgb**
**Proposed changes**:

If two entities are separated by a single space with U- tags, they are currently detected as one entity. 

Example:
my ingredients are eggs lemon juice and milk results in 
{'entity': 'eggs lemon juice',  'entity':: 'milk', 'extractor': 'DIETClassifier'}

This PR fixes this. Entities separated by a single space and BILOU tags will be listed as individual entities.
Related post: https://github.com/RasaHQ/rasa/issues/6340

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @melindaloubser1 will take a look at it as soon as possible ✨

File: rasa/nlu/extractors/extractor.py [hunk 1]
```diff
@@ -148,11 +149,32 @@ def convert_predictions_into_entities(
...
+                            != bilou_utils.bilou_prefix_from_tag(current_entity_tag)
+                            and bilou_utils.INSIDE
+                            != bilou_utils.bilou_prefix_from_tag(current_entity_tag)
+                        )
+                    )
+                    or last_group_tag != current_group_tag
+                    or last_role_tag != current_role_tag
+                    or bilou_utils.UNIT
+                    == bilou_utils.bilou_prefix_from_tag(current_entity_tag)
+                )
```
Comment (tabergma) [inline rasa/nlu/extractors/extractor.py:168]: Can we move this to a separate method and add some comments? Thanks.

File: tests/nlu/extractors/test_extractor.py [hunk 2]
```diff
@@ -96,13 +96,14 @@
         (
             "Amsterdam Berlin and London",
```
Comment (tabergma) [inline tests/nlu/extractors/test_extractor.py:97]: Can you please add some more test cases like this? E.g. "San Francisco Amsterdam, London" or even something with three tokens? Thanks.

File: rasa/nlu/classifiers/diet_classifier.py [hunk 3]
```diff
@@ -935,7 +935,6 @@ def _entity_label_to_tags(
 
             if self.component_config[BILOU_FLAG]:
                 tags = bilou_utils.ensure_consistent_bilou_tagging(tags)
-                tags = bilou_utils.remove_bilou_prefixes(tags)
```
Comment (tabergma) [inline rasa/nlu/classifiers/diet_classifier.py:938]: We should also remove this line from the CRF Entity Extractor (https://github.com/RasaHQ/rasa/blob/master/rasa/nlu/extractors/crf_entity_extractor.py#L289).

File: rasa/nlu/extractors/extractor.py (hunk 1 above)
Comment (tabergma) [reply rasa/nlu/extractors/extractor.py:168]: Or we could split the statement into multiple ones, e.g. 
```
role_or_group_changed = last_group_tag != current_group_tag or last_role_tag != current_role_tag
new_bilou_tag_begins = last_entity_tag != current_entity_tag and (bilou_utils.LAST != bilou_utils.bilou_prefix_from_tag(current_entity_tag) and bilou_utils.INSIDE != bilou_utils.bilou_prefix_from_tag(current_entity_tag) )
new_tag_starts = last_entity_tag == NO_ENTITY_TAG or bilou_utils.UNIT == bilou_utils.bilou_prefix_from_tag(current_entity_tag)

return new_tag_starts or role_or_group_changed or new_bilou_tag_begins
```
Not sure about the names. Maybe you can come up with better ones.

File: rasa/nlu/classifiers/diet_classifier.py (hunk 3 above)
Comment (praneethgb) [reply rasa/nlu/classifiers/diet_classifier.py:944]: Yes, it was removed from CRF extractor as well. Thanks.

File: tests/nlu/extractors/test_extractor.py (hunk 2 above)
Comment (praneethgb) [reply tests/nlu/extractors/test_extractor.py:97]: Added more test cases to cover 1-3 tokens.

File: rasa/nlu/extractors/extractor.py (hunk 1 above)
Comment (praneethgb) [reply rasa/nlu/extractors/extractor.py]: Sure, made the changes. Thanks.

**Comment by tabergma** [comment]
@praneethgb You need to sign the Contributor License Agreement. Otherwise I am not able to merge this PR. Thanks.

**Comment by praneethgb** [comment]
@tabergma, Signed. Thanks.

**Comment by elenaviter** [comment]
When use **roles**, there's still a problem to have the entity tokens B-entity ... L-entity to be collected to the single token because of the condition:

(https://github.com/RasaHQ/rasa/blob/cfe49f4af1a61a07df7774ed60ade5c3edacbf6d/rasa/nlu/extractors/extractor.py#L152)
`group_or_role_changed = (last_group_tag != current_group_tag or last_role_tag != current_role_tag)`
Considering the role and the entity type are both associated with the same token, for me, this works instead 

```
group_or_role_changed = (
                    last_group_tag != current_group_tag or bilou_utils.tag_without_prefix(last_role_tag) != bilou_utils.tag_without_prefix(current_role_tag)
            )
```
And the rest of the logic to reconstruct the compound entity based on bilou labels of the entity tag ensures the integrity of the approach.

Also the "current_role_tag" under construction of the entity -> `bilou_utils.tag_without_prefix(current_role_tag)` :
```
entity = self._create_new_entity(
                    list(tags.keys()),
                    current_entity_tag,
                    current_group_tag,
                    bilou_utils.tag_without_prefix(current_role_tag),
                    token,
                    idx,
                    confidences,
                )
```

**Comment by wochinge** [comment]
@tabergma have you seen the message from above?

**Comment by tabergma** [comment]
It is fixed on master (https://github.com/RasaHQ/rasa/pull/6577) and I just created a [PR](https://github.com/RasaHQ/rasa/pull/6714) for a minor release.

**Pull request merged**